backend/server/controllers/user.js

```diff
@@ -130,34 +130,129 @@ export const getUserActivity = async (req, res) => {
+        const postsActivity = posts.map((p) =>
+            Activity.convertToActivity(Activity.POSTED, p.id, p.createdAt)
+        )
 
         const sharedPosts = await models.sharedPost.findAll({
-            where:{
-                userID: userOfInterest.id
-            }
+            where: {
+                userID: userOfInterest.id,
+            },
         })
 
-        const sharedPostsActivity = sharedPosts.map(p => Activity.convertToActivity(Activity.SHARED, p.postId, p.createdAt))
+        const sharedPostsActivity = sharedPosts.map((p) =>
+            Activity.convertToActivity(Activity.SHARED, p.postId, p.createdAt)
+        )
 
         const likedPosts = await models.likedPost.findAll({
-            where:{
-                userID: userOfInterest.id
-            }
+            where: {
+                userID: userOfInterest.id,
+            },
         })
 
-        const likedPostsActivity = likedPosts.map(p => Activity.convertToActivity(Activity.LIKED, p.postId, p.createdAt))
+        const likedPostsActivity = likedPosts.map((p) =>
+            Activity.convertToActivity(Activity.LIKED, p.postId, p.createdAt)
+        )
 
         // Merge the posts together into one array, sort them in descending order based on timestamp, and return them
-        let activity = [...postsActivity, ...likedPostsActivity, ...sharedPostsActivity];
-        activity.sort((a, b) => (a.timestamp < b.timestamp) ? 1 : -1)
+        let activity = [
+            ...postsActivity,
+            ...likedPostsActivity,
+            ...sharedPostsActivity,
+        ]
+        activity.sort((a, b) => (a.timestamp < b.timestamp ? 1 : -1))
         res.status(200).send(activity)
     } catch (error) {
         res.status(500).send({ 'Error message': error.toString() })
     }
 }
+
+/*
+Require authentication. (only you can see list of follower and followings)
+Path paramter: username - the username of the account we want 
+    to get following and followers for.
+Response Codes:
+200 OK when the followers and following has been successfully found.
+404 NOT FOUND when the post with that id can not be found.
+500 INTERNAL SERVER ERROR for everything else.
+Returns : List of followers and followings.
+*/
+export const getFollow = async (req, res) => {
+    try {
+        const { params } = req
+        const authToken = req.get('Authorization')
+
+        if (!authToken) {
+            res.status(400).send({
+                'Error message': 'Auth token not provided',
+            })
+            return
+        }
+
+        const decodedUser = Authentication.extractUser(authToken)
+        const user = await models.users.findOne({
+            where: { username: params.username },
+        })
+
+        if (!user) {
+            res.status(400).send({
+                'Error message':
+                    'Invalid param : user with given username does not exist',
+            })
+            return
+        }
+
+        if (decodedUser.id === user.id) {
```

Comment:
I don't think we care if decodedUser is the same as user, because the logged user can be viewing any other user's page. Just if (decodedUser.id) should be fine

Reply:
I made it so that only you can see your followers and following. 
I asked whether or not these information should be public or not and Dylan J suggest it that we make it private and can change it to be public later if it needed.
Should I change it to publicly accessible by anyone?

Reply:
OK.
Will keep the authentication but allow anyone with valid account to view people's follower and following list.